### ISSUE TITLE
Error while running make install

### ISSUE THREAD

**Issue by amansinh** [hawk/lux#30]
Hi,
I am trying to install lux on ubuntu 14.04 desktop. Following are the steps that i have followed

1. Installed erlang OTP
   >> http://erlang.org/download/otp_src_20.1.tar.gz
2. Untar the folder and run configure
  >> ./configure
3. Run make
  >> make
4. Run make install
  >> sudo make install

After this otp is installed properly and my erlang run time is working 

After this i clone the current lux repo and execute the following commands

1. autoconf
2. ./configure
3. make
4. sudo make install

The make install command is giving the following error
**sachin@dev-ob:~/lux$ sudo make install 
make[1]: Entering directory `/home/sachin/lux/src'
mkdir -p /usr/local/lux
/home/sachin/lux/bin/lux --install /usr/local/lux
Installing lux as a standalone system... /usr/local/lux
ERROR: Module erl2html2 potentially included by two different applications: common_test and test_server.
make[1]: *** [install] Error 1
make[1]: Leaving directory `/home/sachin/lux/src'
make[1]: Entering directory `/home/sachin/lux/c_src'
make[1]: Nothing to be done for `install'.
make[1]: Leaving directory `/home/sachin/lux/c_src'**

Is there any problem with the steps i am following, or the code is broken. I am using the master branch to install lux

**Comment by hawk** [comment]
Hi,

That looks strange.

It would be interesting to see what this reveals:

erl
>   code:clash().
>   io:format("PATH: ~p\n", [code:get_path()]).
>   halt().
>

​Can you run those commands and send me the output?​

/Håkan

**Comment by amansinh** [comment]
Hi,

Adding the logs from above command

**sachin@dev-ob:~/lux$ erl
Erlang R16B03-1 (erts-5.10.4) [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

Eshell V5.10.4  (abort with ^G)
1> code:clash(). io:format("PATH: ~p\n", [code:get_path()]). halt().
** /usr/local/lib/erlang/lib/typer-0.9.5/ebin/typer.beam hides /usr/local/lib/erlang/lib/dialyzer-3.2.3/ebin/typer.beam
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server_gl.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server_gl.beam
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/erl2html2.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/erl2html2.beam
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server_node.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server_node.beam 
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server_io.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server_io.beam
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server.beam
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server_ctrl.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server_ctrl.beam 
** /usr/local/lib/erlang/lib/test_server-3.6.4/ebin/test_server_sup.beam hides /usr/local/lib/erlang/lib/common_test-1.15.3/ebin/test_server_sup.beam
** Found 8 name clashes in code paths 
ok
2> io:format("PATH: ~p\n", [code:get_path()]). halt().
PATH: [".","/usr/local/lib/erlang/lib/kernel-2.16.4/ebin",
       "/usr/local/lib/erlang/lib/stdlib-1.19.4/ebin",
       "/usr/local/lib/erlang/lib/xmerl-1.3.16/ebin",
       "/usr/local/lib/erlang/lib/wx-1.8.3/ebin",
       "/usr/local/lib/erlang/lib/webtool-0.8.9.2/ebin",
       "/usr/local/lib/erlang/lib/typer-0.9.5/ebin",
       "/usr/local/lib/erlang/lib/tv-2.1.4.10/ebin",
       "/usr/local/lib/erlang/lib/tools-2.11.1/ebin",
       "/usr/local/lib/erlang/lib/toolbar-1.4.2.3/ebin",
       "/usr/local/lib/erlang/lib/test_server-3.6.4/ebin",
       "/usr/local/lib/erlang/lib/syntax_tools-2.1.4/ebin",
       "/usr/local/lib/erlang/lib/ssl-8.2.3/ebin",
       "/usr/local/lib/erlang/lib/ssh-4.6.3/ebin",
       "/usr/local/lib/erlang/lib/snmp-5.2.9/ebin",
       "/usr/local/lib/erlang/lib/sasl-3.1.1/ebin",
       "/usr/local/lib/erlang/lib/runtime_tools-1.12.3/ebin",
       "/usr/local/lib/erlang/lib/reltool-0.7.5/ebin",
       "/usr/local/lib/erlang/lib/public_key-1.5.2/ebin",
       "/usr/local/lib/erlang/lib/pman-2.7.1.4/ebin",
       "/usr/local/lib/erlang/lib/percept-0.8.8.2/ebin",
       "/usr/local/lib/erlang/lib/parsetools-2.1.6/ebin",
       "/usr/local/lib/erlang/lib/otp_mibs-1.1.2/ebin",
       "/usr/local/lib/erlang/lib/os_mon-2.4.4/ebin",
       "/usr/local/lib/erlang/lib/orber-3.8.4/ebin",
       "/usr/local/lib/erlang/lib/odbc-2.12.1/ebin",
       "/usr/local/lib/erlang/lib/observer-2.6/ebin",
       "/usr/local/lib/erlang/lib/mnesia-4.15.2/ebin",
       "/usr/local/lib/erlang/lib/megaco-3.18.3/ebin",
       "/usr/local/lib/erlang/lib/jinterface-1.8.1/ebin",
       "/usr/local/lib/erlang/lib/inets-6.4.5/ebin",
       "/usr/local/lib/erlang/lib/ic-4.4.3/ebin",
       "/usr/local/lib/erlang/lib/hipe-3.17/ebin",
       "/usr/local/lib/erlang/lib/gs-1.5.15.2/ebin",
       "/usr/local/lib/erlang/lib/eunit-2.3.5/ebin",
       "/usr/local/lib/erlang/lib/et-1.6.1/ebin",
       "/usr/local/lib/erlang/lib/erts-9.2/ebin",
       "/usr/local/lib/erlang/lib/erl_interface-3.10.1/ebin",
       "/usr/local/lib/erlang/lib/erl_docgen-0.7.2/ebin",
       "/usr/local/lib/erlang/lib/eldap-1.2.3/ebin",
       "/usr/local/lib/erlang/lib/edoc-0.9.2/ebin",
       "/usr/local/lib/erlang/lib/diameter-2.1.3/ebin",
       "/usr/local/lib/erlang/lib/dialyzer-3.2.3/ebin",
       "/usr/local/lib/erlang/lib/debugger-4.2.4/ebin",
       "/usr/local/lib/erlang/lib/crypto-4.2/ebin",
       "/usr/local/lib/erlang/lib/cosTransactions-1.3.3/ebin",
       "/usr/local/lib/erlang/lib/cosTime-1.2.3/ebin",
       "/usr/local/lib/erlang/lib/cosProperty-1.2.3/ebin",
       "/usr/local/lib/erlang/lib/cosNotification-1.2.3/ebin",
       "/usr/local/lib/erlang/lib/cosFileTransfer-1.2.2/ebin",
       "/usr/local/lib/erlang/lib/cosEventDomain-1.2.2/ebin",
       "/usr/local/lib/erlang/lib/cosEvent-2.2.2/ebin",
       "/usr/local/lib/erlang/lib/compiler-7.1.4/ebin",
       "/usr/local/lib/erlang/lib/common_test-1.15.3/ebin",
       "/usr/local/lib/erlang/lib/asn1-5.0.4/ebin",
       "/usr/local/lib/erlang/lib/appmon-2.1.14.2/ebin"]
ok
3> halt().**

**Comment by hawk** [comment]
Hi,

Thanks for the printouts.

It looks like you have installed an older Erlang system at some point and
installed the new one on top of the old.

I suggest that you do remove the current installation of Erlang and
reinstall it:

    sudo rm -rf /usr/local/lib/erlang
    sudo make install

But this may also work:

    sudo rm -rf /usr/local/lib/erlang/typer\*
    sudo rm -rf /usr/local/lib/erlang/test_server\*

/Håkan

**Comment by amansinh** [comment]
Thanks Hakan, I can see lux installed now. 
I am not sure if i had a previous version of erlang installed in my setup. 
So, just to confirm if "sudo apt-get install erlang" is enough before running the lux installation commands or do we need to specifically build the erlang/otp as i have mentioned in the 1st comment ?

**Comment by hawk** [comment]
No, you do not need to build Erlang/OTP yourself. It should suffice to
install erlang with "sudo apt-get install erlang" . (But you should
probably remove the newly installed erlang before you do that.)

/Håkan

**Comment by amansinh** [comment]
Thanks, it solves the issue. I will be closing the issue.

On 14-Dec-2017 10:21 PM, Håkan Mattsson <notifications@github.com> wrote:
No, you do not need to build Erlang/OTP yourself. It should suffice to
install erlang with "sudo apt-get install erlang" . (But you should
probably remove the newly installed erlang before you do that.)

/Håkan
